tlz/fips.go

```diff
@@ -1,10 +1,13 @@
-//go:build goexperiment.opensslcrypto
+//go:build goexperiment.opensslcrypto && requirefips
```

Comment:
this way the function can only return true if OpenSSL FIPS mode is required/enforced, a build tag that prevents the compiled binary from executing when OpenSSL FIPS Provider is not active